cvat-core/src/api-implementation.js

```diff
@@ -199,11 +209,48 @@
             return tasks;
         };
 
-        cvat.server.installedApps.implementation = async () => {
-            const result = await serverProxy.server.installedApps();
-            return result;
+        cvat.projects.get.implementation = async (filter) => {
+            checkFilter(filter, {
+                id: isInteger,
+                page: isInteger,
+                name: isString,
+                assignee: isString,
+                owner: isString,
+                search: isString,
+                status: isEnum.bind(TaskStatus),
+            });
+
+            if ('search' in filter && Object.keys(filter).length > 1) {
+                if (!('page' in filter && Object.keys(filter).length === 2)) {
+                    throw new ArgumentError('Do not use the filter field "search" with others');
+                }
+            }
+
+            if ('id' in filter && Object.keys(filter).length > 1) {
+                if (!('page' in filter && Object.keys(filter).length === 2)) {
+                    throw new ArgumentError('Do not use the filter field "id" with others');
+                }
+            }
+
+            const searchParams = new URLSearchParams();
+            // TODO: need to check search fields
+            for (const field of ['name', 'assignee', 'owner', 'search', 'status', 'id', 'page']) {
+                if (Object.prototype.hasOwnProperty.call(filter, field)) {
+                    searchParams.set(field, filter[field]);
+                }
+            }
+
+            const projectsData = await serverProxy.projects.getProjects(searchParams.toString());
+            // prettier-ignore
+            const projects = projectsData.map((project) => new Project(project));
+
+            projects.count = projectsData.count;
+
+            return projects;
         };
 
+        cvat.projects.searchNames.implementation = async (search, limit) => serverProxy.projects.searchProjectNames(search, limit);
```

Comment:
Why not implement it with ``get`` (according to users, just provide BasicProjectSerializer that includes name and id)?

Reply:
It was done like you say, but I used different method because `get` method returns `Project` object and this method returns only `{id: ..., name: ...}` structure